sound/soc/intel/boards/sof_cirrus_common.c

```diff
@@ -168,6 +168,11 @@ static int cs35l41_compute_codec_conf(void)
 			continue;
 		}
 		physdev = get_device(acpi_get_first_physical_node(adev));
+		if (!physdev) {
+			pr_devel("Cannot find physical node for HID %s UID %u (%s)\n", CS35L41_HID,
+					uid, cs35l41_name_prefixes[uid]);
+			continue;
+		}
```

Comment:
The classic pattern is

````c
	codec_dev = acpi_get_first_physical_node(adev);
	acpi_dev_put(adev);
	if (!codec_dev)
		return -EPROBE_DEFER;

        priv->codec_dev = get_device(codec_dev);
````

Your code is leaking one reference here.

Reply:
additional nit-pick: the prefix should be ASoC: Intel: sof_cirrus_common:

Reply:
@plbossart are your sure? `acpi_get_first_physical_node` which calls `acpi_dev_get_next_match_dev` calls `acpi_dev_put`

Reply:
That being said there is a very confusing note attached

Reply:
ah nvm i see i got my adevs crossed